### DISCUSSION TITLE
Need to calculate Mercury Phase Angle

### DISCUSSION THREAD

**Discussion by jaclerigo** [DeepskyLog/laravel-astronomy-library#88] [Q&A]
Hello,

anyone knows how to calculate Mercury Phase Angle?
Can't find a way to do it.

**Comment by WimDeMeester** [comment]
Will the method

illuminatedFraction($date)

not give you the correct answer?

Can you tell me a bit more about the project you are implementing (I'm interested in use cases for the astronomy library)?

**Reply by jaclerigo** [reply]
Hi there,

I'm trying to create a simple image of the mercury phase as seen from Earth.
I really don't know if I'm doing it right, but I think that I need 3 parameters for this.

1. the date/time
2. the $mercury->illuminatedFraction($date);
3.  and Mercury angle phase or elongation (I guess that elongations is best)

But I could not figure out how to get one or the other.

About the project :) not really nothing much, I'm just an amateur astronomer that likes programming too. Just found out about this library because I was interested in creating a sort of visual phases calendar for Mercury and Venus.

**Reply by WimDeMeester** [reply]
I don't think the angle phase can be calculated at the moment.  I need to check my copy of Astronomical Algorithms to see if there is a formula for that.  You are also always welcome to make a pull request if you succeed in adding this functionality.

**Comment by jaclerigo** [comment]
I see that there is $mercury->greatest_western_elongation($date) and $eastern = $mercury->greatest_eastern_elongation($date);
Is it possible to obtain only the elongation for a certain date?

**Reply by WimDeMeester** [reply]
That is not possible at the moment...  If you look at the code at https://github.com/DeepskyLog/laravel-astronomy-library/blob/master/src/deepskylog/AstronomyLibrary/Targets/Mercury.php#L378 you can see that the eastern elongation is calculated by one formula, but not by calculating the elongation itself...
